### ISSUE TITLE
Identify when 2 func arguments of the same type next to one another 

### ISSUE THREAD

**Issue by briandowns** [golang/lint#267]
Identify when 2 func arguments of the same type next to one another for the purpose of suggesting that they be rewritten with a single type identifier.  I have the first pass for this working and will continue to clean it up if it's deemed worthwhile.

**Comment by dsnet** [comment]
I have seen some legitimate bugs because of accidental swapping of the arguments. But the check your proposing will also have many false positives on functions like: `func Max(a, b int)`

**Comment by briandowns** [comment]
Given 
```
func Max(x int, y int) {}
```
It would identify that but not identify the above example.

**Comment by dsnet** [comment]
Oh sorry, I misread your original suggestion.

I think this seems reasonable. If you want to write it up and see what things it catches, we can see what things it flags. If it flags things that are reasonable, we should avoid this check, but nothing significant comes to mind.

**Comment by andybons** [comment]
Thanks for the issue.

This appears to be [out of scope](https://github.com/golang/lint/blob/master/README.md#scope) and so is being closed. If you feel this is in error please feel free to re-open.

Please note this is not the forum to discuss edits to [CodeReviewComments](https://golang.org/wiki/CodeReviewComments). If you wish to propose changes to it please send a mail to [golang-nuts](https://groups.google.com/forum/#!forum/golang-nuts).
